src/compiler/checker.ts

```diff
@@ -25321,7 +25342,10 @@ namespace ts {
                 }
 
                 if (computedNameType && !(computedNameType.flags & TypeFlags.StringOrNumberLiteralOrUnique)) {
-                    if (isTypeAssignableTo(computedNameType, stringNumberSymbolType)) {
+                    if (isTypeAny(computedNameType)) {

```

Comment:
Technically, I can push this change through without this change, if we'd prefer. However, that means when you write, eg
```ts
const obj = {
  [Symbol.whatever]: 0
};
```
`obj` would get a `number` index signature (since `Symbol.whatever` is going to be `any` from the error of it not existing), whereas with this it gets a `string` index signature (which, IMO, makes much more sense for an `any` computed name). It mostly only matters, in the context of unique symbols, for follow-on errors to nonexistant symbols which were previously "well known" and thus entirely unchecked.

Reply:
From the comment here, it sounds like this mostly to cater to the "degenerate" case of `Symbol.thingThatDoesNotExist` which can happen in a loosely configured JavaScript project - is that the use-case you have in mind? And 

That does loosen the behavior of this by quite a bit. I think I would be willing to go the *opposite* way and take a break on this behavior.

Reply:
You definitely don't wanna go the opposite way, because the numeric index signature we otherwise make makes very little sense, IMO. I actually think `any` producing a `number` index signature just feels a bit like a bug that we aight to fix.

Reply:
I get that, but I think the `any` behavior is already pretty undesirable, and is exacerbated if you're not using `--noUncheckedIndexedAccess`.

Reply:
???? That flag is about accessing keys and the types accessed keys are assigned, where here the issue is solely weather or not we make a key at all, and what slot that key should go in; that seems wholly unrelated to that flag.

Reply:
Sorry, I meant `--noPropertyAccessFromIndexSignature`.

Reply:
> I get that, but I think the any behavior is already pretty undesirable

Yes! Which is why this improves it, so
```ts
const s = {["str" as any]: "str"}
```
s can be indexed by `"str"` rather than only objective nonsense whose lookup would never work like `12`. And let's be perfectly clear here: you've used an `any` (or have one derived from an error) - _some part_ of type checking is going to be turned off. We shouldn't _ignore_ an `any`-typed computed property key (and I'm definitely not going to do that in this PR, that's for sure - that'd be a massive breaking change) - then you'd just end up with `{}` which is _really_ going to hurt a _lot_ of common uses.

Reply:
But

```ts
const s = {["str" as any]: "str"}
```

can now be indexed by a wider variety of nonsense that would never work either, like `s["uh oh"]`. You're allowing a much more optimistic type to be formed which you can't take back after the fact.

Reply:
And in fact, I think that's why the numeric index signature was chosen - because at the time we also felt the behavior it supported was *more* desirable even if it wasn't good.

The point of the numeric index signature isn't to support indexing with `number`s - it's to support indexing with `any` and still getting the appropriate type out. It's a hack, but it doesn't widen the net to every possible non-numeric property.

Reply:
But the point of `any` _is_ to "widen the net".

Reply:
At best this isn't something we need to change with this feature. I don't find the "well known symbol that doesn't exist" thing to be a use-case that should fundamentally change what happens when you use `any` as a computed property key.

Reply:
Alright 🤷‍♂️